jobspec/parse_service.go

```diff
@@ -49,6 +50,7 @@ func parseService(o *ast.ObjectItem) (*api.Service, error) {
 		"task",
 		"meta",
 		"canary_meta",
+		"on_update",
```

Comment:
We should not be adding this to the legacy HCL1 stuff, right?
https://github.com/hashicorp/nomad/blob/master/contributing/checklist-jobspec.md#hcl1-deprecated

Reply:
deprecated but still supported I guess I thought? :thinking: